models/schema.go

```diff
@@ -9,28 +9,75 @@ type Schema struct {
 	Sites []map[string]interface{} `json:",omitempty"`
 }
 
-func NewSchema(id, displayName, templateName, tenantId string) *Schema {
-	templateMap := map[string]interface{}{
-		"name":          templateName,
-		"tenantId":      tenantId,
-		"displayName":   templateName,
-		"anps":          []interface{}{},
-		"contracts":     []interface{}{},
-		"vrfs":          []interface{}{},
-		"bds":           []interface{}{},
-		"filters":       []interface{}{},
-		"externalEpgs":  []interface{}{},
-		"serviceGraphs": []interface{}{},
+type SchemaReplace struct {
+	Ops   string                   `json:",omitempty"`
+	Path  string                   `json:",omitempty"`
+	Value []map[string]interface{} `json:",omitempty"`
+}
+
+func NewSchema(id, displayName, templateName, tenantId string, template []interface{}) (*Schema, *SchemaReplace) {
+	result := []map[string]interface{}{}
+	if templateName != "" {
+		templateMap := map[string]interface{}{
+			"name":          templateName,
+			"tenantId":      tenantId,
+			"displayName":   templateName,
+			"anps":          []interface{}{},
+			"contracts":     []interface{}{},
+			"vrfs":          []interface{}{},
+			"bds":           []interface{}{},
+			"filters":       []interface{}{},
+			"externalEpgs":  []interface{}{},
+			"serviceGraphs": []interface{}{},
+		}
+		result = []map[string]interface{}{
+			templateMap,
+		}
+	} else {
+		templateMap := make(map[string]interface{})
+		for _, map_values := range template {
+			map_template := make(map[string]interface{})
```

Comment:
Could this be initialized as the list in 45 so you don't need to merge it later?

Reply:
Yes..Done